tests/autorestart/test_container_autorestart.py

```diff
@@ -321,9 +321,16 @@ def postcheck_critical_processes_status(duthost, container_autorestart_states, u
     Returns:
       True if post check succeeds; Otherwise False.
     """
-    for container_name in container_autorestart_states.keys():
-        if is_hiting_start_limit(duthost, container_name):
-            clear_failed_flag_and_restart(duthost, container_name)
+    for service in container_autorestart_states.keys():
+        if service in duthost.DEFAULT_ASIC_SERVICES:
+            for asic in duthost.asics:
+                service_name = asic.get_service_name(service)
```

Comment:
I checked the function definition of `get_service_name(...)` at https://github.com/Azure/sonic-mgmt/blob/26422604d185d28143e9cd442503ed4eeed2699c/tests/common/devices/sonic_asic.py#L189. 

Actually the `service_name` at here represents the prefix of systemd configuration file name such as `lldp@0.service`.
Wondering can we use a more specific name such as `systemd_service_name` to replace `service_name` ?

Reply:
We use service_name in most places, for clarity, modified the container_autorestart_states - to feature_autorestart_states. This will have the feature autorestart states from "show feature status". 
For feature swss, service_name will be swss@0 and docker_name will swss0 for asic0.
Added comment to explain this.

Reply:
I think we need reach an agreement on the definition of `feature_name`, `container_name` and `service_name` and formalize usage of these variables.

Reply:
Might be hard to do a single sweep fix, could open a github issue to check other test cases and see if fix is required in the test case or libraries.